### ISSUE TITLE
Sign in with sms_mfa challenge on Cognito sends otp again

### ISSUE THREAD

**Issue by Deepakms92** [aws/aws-sdk-java#2018]
I have added the MFA for a user on Cognito console and when i do a login through my java sdk it finds a SMS_MFA challenge , so i get an OTP when i login . However i give the OTP and try to login and again i get a new OTP ,as a result of which i'm getting Codemismatch Exception. i have given the code here .is it and issue from AWS side?

I Have added the code that handles the challenge along with the code that does the login

` public SpringSecurityUser signIn(AuthenticateRequest authenticateRequest) {
        AuthenticationResultType authenticationResult = null;
        AWSCognitoIdentityProvider cognitoClient = getAWSCognitoIdentityProvider();

        try {
            final Map<String, String> authParams = new HashMap<>();
            authParams.put(USERNAME, authenticateRequest.getUsername());
            authParams.put(PASS_WORD, authenticateRequest.getPassword());

            String sessionResult= null;
            final AdminInitiateAuthRequest authRequest = new AdminInitiateAuthRequest();
            authRequest.withAuthFlow(AuthFlowType.ADMIN_NO_SRP_AUTH)
                    .withClientId(clientId)
                    .withUserPoolId(userPoolId)
                    .withAuthParameters(authParams);

            AdminInitiateAuthResult result = cognitoClient.adminInitiateAuth(authRequest);
            if (StringUtils.isNotBlank(result.getChallengeName())) {
                if (SMS_MFA.equals(result.getChallengeName())) {
                    if (null == authenticateRequest.getPassword()) {
                        throw new CognitoException("User must provide a new password", CognitoException.USER_MUST_CHANGE_PASS_WORD_EXCEPTION_CODE, result.getChallengeName());
                    } else {

                        final Map<String, String> challengesResponse = new HashMap<>();
                        challengesResponse.put(USERNAME, authenticateRequest.getUsername());
                        challengesResponse.put(PASS_WORD, authenticateRequest.getPassword());
                        challengesResponse.put(SMS_MFA_CODE, authenticateRequest.getMfaCode());
                        final AdminRespondToAuthChallengeRequest requestChallenge = new AdminRespondToAuthChallengeRequest();
                        requestChallenge
                                .withChallengeName(ChallengeNameType.SMS_MFA)
                                .withChallengeResponses(challengesResponse)
                                .withClientId(clientId)
                                .withUserPoolId(userPoolId)
                                .withSession(result.getSession());

                        AdminRespondToAuthChallengeResult requestChallengeResponse = cognitoClient.adminRespondToAuthChallenge(requestChallenge);
                        authenticationResult = requestChallengeResponse.getAuthenticationResult();

                    }

                } else {
                    throw new CognitoException(result.getChallengeName(), CognitoException.USER_MUST_DO_ANOTHER_CHALLENGE, result.getChallengeName());
                }

            } else {

                authenticationResult = result.getAuthenticationResult();
            }
            SpringSecurityUser userAuthenticated = new SpringSecurityUser(authenticateRequest.getUsername(), authenticateRequest.getPassword(), authenticateRequest.getMfaCode(), null, null, null);
            userAuthenticated.setAccessToken(authenticationResult.getAccessToken());
            userAuthenticated.setExpiresIn(authenticationResult.getExpiresIn());
            userAuthenticated.setTokenType(authenticationResult.getTokenType());
            userAuthenticated.setRefreshToken(authenticationResult.getRefreshToken());
            userAuthenticated.setIdToken(authenticationResult.getIdToken());
            log.info("User authenticated" + authenticateRequest.getUsername());

            return userAuthenticated;
        } catch (AWSCognitoIdentityProviderException awse) {
            throw new CognitoException(awse.getMessage(), awse.getErrorCode(), awse.getMessage() + awse.getErrorCode());

        } catch (CognitoException cognitoException) {
            throw cognitoException;
        } catch (Exception e) {
            throw new CognitoException(e.getMessage(), CognitoException.GENERIC_EXCEPTION_CODE, e.getMessage());
        }

    }`

**Comment by debora-ito** [comment]
Hi @Deepakms92 I could not reproduce your issue. We have an older issue reporting the same problem (#1914), and I tried the same steps again, which I'm listing here:

1.  Created a new user `user01` using AWS Console, provided phone number and email. User status is FORCE_CHANGE_PASSWORD
2. Using the Java SDK, I initiated a new `AuthFlow` of type ADMIN_NO_SRP_AUTH for `user01` , which returned a challenge type NEW_PASSWORD_REQUIRED
3. In a `RespondToAuthChallengeRequest`, I provided username `user01`, a new password and challenge type NEW_PASSWORD_REQUIRED. As a result I got a new challenge SMS_MFA
4. Then I used a snippet of your code to provide `user01`, password (the new one) and the SMS_MFA_CODE I got from SMS, in a `AdminRespondToAuthChallengeRequest`. 

But after that I did not get a new code, as you did. 

Since this is a question specific to Cognito and its authentication flows, I'd suggest to post your question in the  [AWS Developer Forum](https://forums.aws.amazon.com/forum.jspa?forumID=173)  to get the service support.
